subgraphs/uniswap-v2/src/Factory.ts

```diff
@@ -0,0 +1,150 @@
+// import { log } from '@graphprotocol/graph-ts'
+import { log } from '@graphprotocol/graph-ts'
+import { PairCreated } from './../generated/Factory/Factory'
+import { DexAmmProtocol, LiquidityPool, Token, _HelperStore, _TokenTracker } from './../generated/schema'
+import { Pair as PairTemplate } from '../generated/templates'
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from './common/tokens'
+import { WHITELIST } from './common/Price'
+import {
+  FACTORY_ADDRESS,
+  BIGDECIMAL_ZERO,
+  ProtocolType,
+  Network,
+  BIGDECIMAL_ONE,
+  INT_ZERO
+} from './common/constants'
+
+export function handleNewPair(event: PairCreated): void {
+
+  log.debug('New Pair Created -', [])
+
+  let protocol = DexAmmProtocol.load(FACTORY_ADDRESS)
+
+  if (protocol === null) {
+    protocol = new DexAmmProtocol(FACTORY_ADDRESS)
+    protocol.name = "Uniswap v2"
+    protocol.slug = "uniswap-v2"
+    protocol.totalValueLockedUSD = BIGDECIMAL_ZERO
+    protocol.network = Network.ETHEREUM
+    protocol.type = ProtocolType.EXCHANGE
+
+    // create new ether price storage object
+    let ether = new _HelperStore('ETH')
+    ether.valueDecimal = BIGDECIMAL_ZERO
+    ether.save()
+
+    // Tracks the total number of unique users of the protocol 
+    let uniqueUsersTotal = new _HelperStore('USERS')
+    uniqueUsersTotal.valueInt = INT_ZERO
+    uniqueUsersTotal.save()
+  }  
+
+  // create the tokens
+  let token0 = Token.load(event.params.token0.toHexString())
+  let token1 = Token.load(event.params.token1.toHexString())
+  let LPtoken = Token.load(event.params.pair.toHexString())
+
+  // fetch info if null
+  if (token0 === null) {
+    token0 = new Token(event.params.token0.toHexString())
+    token0.symbol = fetchTokenSymbol(event.params.token0)
+    token0.name = fetchTokenName(event.params.token0)
+    let decimals = fetchTokenDecimals(event.params.token0)
+
+    // bail if we couldn't figure out the decimals
+    if (decimals == 9999) {
+      log.debug('mybug the decimal on token 0 was null', [])
+      return
+    }
+
+    token0.decimals = decimals
+  }
+
+  if (token1 === null) {
+    token1 = new Token(event.params.token1.toHexString())
+    token1.symbol = fetchTokenSymbol(event.params.token1)
+    token1.name = fetchTokenName(event.params.token1)
+    let decimals = fetchTokenDecimals(event.params.token1)
+    
+    // bail if we couldn't figure out the decimals
+    if (decimals == 9999) {
+      log.debug('mybug the decimal on token 0 was null', [])
```

Comment:
Also, im wondering, how often does this occur? Can you share some example tokens where this occurs?

Reply:
I'm trying to decide if it's a good idea to not index the pairs containing a token without decimals, or have some other ways to handle them (e.g. use a default 18)

Reply:
This was code from the other subgraph. I assumed the previous programmer had a good reason for it. I am running the subgraph now without the null check to see if anything happens.